### DISCUSSION TITLE
Test Continuation after Reboot

### DISCUSSION THREAD

**Discussion by jpavlav** [avocado-framework/avocado#4426] [Q&A]
Hey folks! I'm not 100% sure if this is the best place to ask this kind of question but hey, "Q&A" 😄 . If Avocado is running a test on a server and the server does a reboot as part of the test, are there any mechanisms that would allow for the continuation of the test after the server is back up?

**Comment by clebergnu** [comment]
Hi @jpavlav ,

This feature is one that we have not yet "ported" from the Autotest framework, where it was common to build kernels, reboot into them, and continue the test.

It's also not on the immediate development plans (see https://github.com/avocado-framework/avocado/projects/1 ).  Having said that, if you'd like to investigate and invest some time into it, I'd more than willing to split the effort given that I'm familiar with the Autotest feature and that will certainly help.

Cheers!

**Reply by jpavlav** [reply]
Hello @clebergnu ,

Thank you for the quick response! It's definitely something I'd be interested in looking into. Can you point me in the right direction as far as where to look in the Autotest codebase? It's quite large 😄 . Maybe we can get something going for this in Avocado.

Thanks!

**Reply by clebergnu** [reply]
This is a documentation on how to use the feature:

https://github.com/autotest/autotest/blob/master/documentation/source/main/local/ControlHowto.rst#rebooting-during-a-job

And hopefully it gives hints on where the bits and pieces are implemented.  There's code that handles the various steps of a job (including setting what's the next step to be run, in this case after the machine reboots), and the preservation of such state (look for `control.state`) to be loaded and resumed by the "autotest client".

Now, while it's useful to look at this implementation, don't get too attached as they are very specific to the Autotest architecture, and Avocado will probably differ quite a bit.
